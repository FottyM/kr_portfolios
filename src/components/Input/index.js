import React from 'react'
import PropTypes from 'prop-types'

const Input = ({ id, type, name, value, onChange, label, placeholder }) => {
  return (
    <div className="form-group">
      <label htmlFor={id}>{label}</label>
      <input
        type={type}
        className="form-control"
        name={name}
        id={id}
        placeholder={placeholder}
        onChange={onChange}
        value={value} />
    </div>
  )
}

Input.propTypes = {
  name: PropTypes.string.isRequired,
  type: PropTypes.oneOf(['text', 'email', 'password']),
  id: PropTypes.string.isRequired,
  placeholder: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired
}

Input.defaultProps = {
  type: 'text',
  placeholder: '',
}

export default Input 