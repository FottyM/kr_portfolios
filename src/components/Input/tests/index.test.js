import React from 'react'
import { shallow } from 'enzyme'

import Input from '../index'

describe('<Input />', () => {
  it('renders without crashing', () => {
    const spy = sinon.spy()
    const component = shallow(<Input id={'id'} label={'label'} name={'name'} onChange={spy} value={'value'} />)
    expect(component).toMatchSnapshot()
  })
})
