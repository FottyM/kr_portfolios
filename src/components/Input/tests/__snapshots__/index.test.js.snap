// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders without crashing 1`] = `
<div
  className="form-group"
>
  <label
    htmlFor="id"
  >
    label
  </label>
  <input
    className="form-control"
    id="id"
    name="name"
    onChange={[Function]}
    placeholder=""
    type="text"
    value="value"
  />
</div>
`;
